src/utils/errors.ts

```diff
@@ -3,3 +3,12 @@ export class UnknownNetworkError extends Error {
     super(message);
   }
 }
+
+export class MetaMaskError extends Error {
```

Comment:
This is a bit misleading - metamask errors arent instances of this class. I would reduce it to an interface